tools/cachegrand-benchmark-generator/src/support.c

```diff
@@ -0,0 +1,54 @@
+/**
+ * Copyright (C) 2018-2022 Vito Castellano
+ * All rights reserved.
+ *
+ * This software may be modified and distributed under the terms
+ * of the BSD license.  See the LICENSE file for details.
+ **/
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+#include "support.h"
+
+char* support_read_file(
+        char *filename) {
+    char *buffer = NULL;
+    int string_size, read_size;
+    FILE *handler = fopen(filename, "r");
+
+    if (handler) {
+        fseek(handler, 0, SEEK_END);
+        string_size = ftell(handler);
+        rewind(handler);
+
+        buffer = (char*) malloc(sizeof(char) * (string_size + 1) );
+        read_size = fread(buffer, sizeof(char), string_size, handler);
+        buffer[string_size] = '\0';
```

Comment:
Should ensure that read_size and string_size match, probably it's better to have it in a loop like

```
size_t read_size = 0;
do {
    size_t read_size_temp = fread(buffer, sizeof(char), string_size, handler);
    if (read_size_temp <= 0) {
        perror('Failed to read file');
    }
    read_size += read_size_temp;
} while(read_size < string_size);
```